bxin/ptg2hex/hex_diagnostics.ipynb

```diff
@@ -0,0 +1,607 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## This notebook check the hexapod state transitions and move/offset commands\n",
+    "## It also checks the LUT against the input polynomials, before and after a slew\n",
+    "\n",
+    "This notebook works with both hexapods.\n",
+    "\n",
+    "To switch between the hexapods, change the cell below."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "hexIdx = 1 #camera hexapod\n",
+    "#hexIdx = 2 #M2 hexapod"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from lsst.ts import salobj\n",
+    "import asyncio\n",
+    "import os\n",
+    "import yaml\n",
+    "\n",
+    "import numpy as np\n",
+    "from matplotlib import pyplot as plt\n",
+    "from astropy.time import Time\n",
+    "from datetime import datetime, timedelta\n",
+    "import pandas as pd\n",
+    "\n",
+    "from lsst.ts.idl.enums import ATPtg\n",
+    "from astropy.coordinates import AltAz, ICRS, EarthLocation, Angle, FK5\n",
+    "import astropy.units as u\n",
+    "\n",
+    "#import os\n",
+    "#print(os.environ[\"OSPL_URI\"])\n",
+    "#print(os.environ[\"LSST_DDS_ALIGNER\"]) #this should give an keyError. If not, something wrong with the DDS"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "start_time = datetime.now()\n",
+    "script = salobj.Controller(\"Script\", index=1)\n",
+    "await asyncio.sleep(10) #wait 10 second may help with DDS problems; closing all other kernels may help too\n",
```

Comment:
I will get Tiago's input as that wait was recommended by him.

Reply:
This is needed if the notebook is not attaching to a dds daemon. We need to give the Domain some time to startup before we start plugging in readers or it will cause timeout issues.

Reply:
OK. As Tiago explained to me in more detail: if you are running single process on a system as large as ours, you create a domain and immediately start populating it with readers and writers it will fail some times… It is pretty much a race condition between when the domain register the master and the readers are registered…

Please consider adding a bit more information about the reason for the wait (a more detailed comment). I suspect it will surprise others. I'm a bit surprised we can get away at all with single process mode on this system.